daal4py/sklearn/neighbors/knn.py

```diff
@@ -163,6 +202,65 @@ def kneighbors(self, X=None, n_neighbors=None, return_distance=True):
 
 
 class KNeighborsClassifier(BaseKNeighborsClassifier, KNeighborsMixin):
+    def fit(self, X, y):
+        X_incorrect_type = isinstance(X, NeighborsBase) or isinstance(X, BallTree) or isinstance(X, KDTree)
```

Comment:
Minor: apply recommendation of CodeFactor
`Consider merging these isinstance calls to isinstance(X, (BallTree, KDTree, NeighborsBase)) (consider-merging-isinstance)`

Reply:
Fixed